hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactBooleanFieldTest.java

```diff
@@ -0,0 +1,205 @@
+import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.internal.serialization.impl.DefaultSerializationServiceBuilder;
+import com.hazelcast.nio.serialization.compact.CompactReader;
+import com.hazelcast.nio.serialization.compact.CompactSerializer;
+import com.hazelcast.nio.serialization.compact.CompactWriter;
+import com.hazelcast.test.HazelcastParametrizedRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static com.hazelcast.internal.serialization.impl.compact.CompactTestUtil.createSerializationService;
+
+@RunWith(HazelcastParametrizedRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class CompactBooleanFieldTest {
+    private final SchemaService schemaService = CompactTestUtil.createInMemorySchemaService();
+
+    @Parameterized.Parameter
+    public int itemCount;
+
+
+    @Parameterized.Parameters(name = "itemCount:{0}")
+    public static Object[] parameters() {
+        return new Object[]{0, 1, 8, 10, 100, 1000};
+    }
+
+    private static boolean[] getBooleans(int count) {
+        boolean[] booleans = new boolean[count];
+        Random r = new Random();
+        for (int i = 0; i < count; i++) {
+            booleans[i] = r.nextInt(10) % 2 == 0;
+        }
+        return booleans;
+    }
+
+    @Test
+    public void testBooleanArrayWithCustomSerializer() {
+        CompactSerializationConfig compactSerializationConfig = new CompactSerializationConfig();
+        compactSerializationConfig.addSerializer(new BoolArrayDTOSerializer());
+        SerializationService serializationService = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig))
+                .build();
+        boolean[] bools = getBooleans(itemCount);
+        BoolArrayDTO expected = new BoolArrayDTO(bools);
+
+        Data data = serializationService.toData(expected);
+        BoolArrayDTO actual = serializationService.toObject(data);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testBooleanArrayWithReflectiveSerializer() {
+        SerializationService serializationService = createSerializationService(schemaService);
+        boolean[] bools = getBooleans(itemCount);
+        BoolArrayDTO expected = new BoolArrayDTO(bools);
+
+        Data data = serializationService.toData(expected);
+        BoolArrayDTO actual = serializationService.toObject(data);
+
+        assertEquals(expected, actual);
+    }
```

Comment:
I wrote that in the private messages but writing here as well to make it more visible. Maybe if İhsan agrees as well, we can remove them.

We don't really need to have tests with the reflective serializer and custom serializer. More or less, the reflective serializer is a custom serializer generated on the fly by us. It is using the same APIs as the normal serializers. 

(Not to mention that we have another reflective serializer for Java records that hasn't been tested in this PR, and some tests which are not tested with reflective serializer in this PR like `testMultipleBoolFields`. If it is OK to not test them with both of the reflective serializers, then I believe it should be OK to not test them at all. We already have a proper test suite for the reflective serializers)

Reply:
@ihsandemir can you comment on this? If it is better remove reflective serializer versions of these tests, I will remove them. 

My reasoning was that there is code that is not commonly used like `ReaderWriterAdapter`, and testing both custom and reflective serializers may make sure that we don't do a mistake in that non-common code. However, adding reflective serializer versions for all tests seems cumbersome and non-necessary to me. So maybe we should remove all of them. 

After removing, we should make sure public coverage is still 100%. However, I think not testing with reflective serializer should not affect public coverage.

Reply:
@mdumandag Can you point the test that you mention at this comment: ```We already have a proper test suite for the reflective serializers```

Reply:
@ihsan please search for `reflect` keyword in the compact tests directory, also the `RecordSerializationTest`, and `RecordSerializationIntegrationTest`. These are enough to make sure that the reflective serializers are working correctly for all types. The rest, like tested in this PR, is more about the CompactReader and Writer, which every compact serializer (including the reflective ones) built upon

Reply:
So should I remove every reflective versions of all tests i added? there are 4 tests I added.

Reply:
I believe it is not needed, and if we are going to add tests for them, let's add tests properly. Not leave any tests untested for both of the reflective serializers. 

But I leave the decision to İhsan.

Reply:
OK, lets remove the tests as suggested from here.